modules/baremetal/main.tf

```diff
@@ -0,0 +1,36 @@
+resource "ibm_is_bare_metal_server" "bms" {
+  profile     = var.profile
+  name        = var.prefix
+  image       = var.image
+  zone        = var.zone
+  keys        = var.ssh_key_id
+  vpc         = var.vpc_id
+  bandwidth   = var.bandwidth
+  access_tags = var.access_tags
+
+  # Use primary_network_interface if no VLANs are needed
+  dynamic "primary_network_interface" {
```

Comment:
For VSI, this is the "legacy" way of doing the network interfaces. The "new" preferred way is to always create a VNI and then attach.

I'm not sure if that also applies to bare metal, just meantioning it in case.